src/bindings/python/src/openvino/frontend/pytorch/decoder.py

```diff
@@ -36,18 +37,17 @@ def ivalue_to_constant(ivalue):
         assert ov_type.is_static(), "Can't deduce type for list"
         return op.Constant(ov_type, Shape([len(ivalue)]), ivalue).outputs()
 
-    if isinstance(ivalue, torch.Tensor) and ivalue.type() in pt_to_ov_type_map:
-        try:
-            ovshape = PartialShape(ivalue.size())
-            ovtype = pt_to_ov_type_map[ivalue.type()]
-            ov_const = op.Constant(ovtype, ovshape.get_shape(), ivalue.data_ptr())
-        except Exception:
-            # old variant that makes a slow data copying
-            warnings.warn("[ WARNING ] Constant wasn't able to convert from data_ptr.")
-            nvalues = ivalue.numpy()
-            ovtype = np_to_ov_type_map[str(nvalues.dtype)]
-            ovshape = PartialShape(nvalues.shape)
-            ov_const = op.Constant(ovtype, ovshape.get_shape(), nvalues.flatten().tolist())
+    if isinstance(ivalue, torch.Tensor):
+        if ivalue.ndim == 0:
+            assert str(ivalue.dtype()) in pt_to_ov_type_map, "Type is not known"
+            ov_type = pt_to_ov_type_map[str(ivalue.dtype)]
+            ov_const = op.Constant(ov_type, Shape([]), [ivalue.item()])
+        else:
+            ivalue = ivalue.to(memory_format=torch.contiguous_format)
+            narr = ivalue.numpy(force=True)
+            if not narr.flags['C_CONTIGUOUS']:
+                narr = np.ascontiguousarray(narr)
+            ov_const = op.Constant(narr, shared_memory=True)
```

Comment:
```suggestion
            narr = ivalue.numpy(force=True)
            if not narr.flags['C_CONTIGUOUS']:
                narr = np.ascontiguousarray(narr)
            ov_const = op.Constant(narr, shared_memory=True)
```
Isn't later `ascontiguousarray` just enough in this case? Is Torch making it impossible to detach later without `.to()` call?

Same comment for later usage.

Reply:
`.to(memory_format=torch.contiguous_format)` has nothing to do with `C_CONTIGUOUS` tensors. This is to work with models in `CHANNELS_LAST` memory format: https://pytorch.org/tutorials/intermediate/memory_format_tutorial.html

Reply:
Got it! Thanks!